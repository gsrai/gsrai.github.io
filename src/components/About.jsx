import React from 'react';
import PropTypes from 'prop-types';

export default class About extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className={'about-container'}>
        <div className={'about-top'}>
          <div className={'about-img-container'}>
            <img src="assets/profilepic.jpeg" alt="some image"/>
          </div>
          <div className={'about-stats'}>
            <h5>{this.props.name}</h5>
            <h5>{this.props.age}</h5>
            <h5>{this.props.occupation}</h5>
            <h5>{this.props.location}</h5>
          </div>
        </div>
        <div className={'about-main-content'}>
          <div className={'about-summary'}>
            <p>
            Hi, My name is Gagondeep and I am a Software Developer living in London.
            <br/><br/>
            I specialize in Java and Javascript FullStack development, but in my spare
            time I enjoy learning new technologies and concepts like the blockchain and machine learning.
            <br /><br />Recently in my own time I have been using Javascript to create a few Apps such as a
             mobile game in React Native and a chrome extension in React & Redux. Please check my Github account below
             to see my open source projects.
            </p>
          </div>
          <div className={'about-social-links'}>
            <ul>
              <li><a href="mailto:gagondeep.srai@gmail.com">Email</a></li>
              <li><a href="https://github.com/gsrai">Github</a></li>
              <li><a href="https://uk.linkedin.com/pub/gagondeep-srai/97/a22/265">Linkedin</a></li>
            </ul>
          </div>
        </div>
      </div>
    );
  }
}

About.propTypes = {
  name: PropTypes.string.isRequired,
  age: PropTypes.string.isRequired,
  occupation: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired
};