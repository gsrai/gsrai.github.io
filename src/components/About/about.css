.about-container {
  margin-left: auto;
  margin-right: auto;
  width: 470px;
  background-color: white;
}

.about-container h5 {
  margin: 0;
}

.about-stats {
  display: inline-block;
  vertical-align: top;
  padding-top: 32px;
  padding-left: 5px;
}

.about-img-container {
  display: inline-block;
  padding-left: 12px; 
  padding-top: 10px; 
}

.about-img-container img {
  width: 100px;
  height: 100px;
  border-radius: 50px;
}

.about-social-links > ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
}

.about-social-links li {
  float: right;
  text-align: center;
  vertical-align: middle;
}

.about-social-links a {
  display: block;
  color: black;
  text-align: center;
  padding: 8px 12px;
  text-decoration: none;
}

.about-social-links a:hover {
  background-color: #ffffd7;
}

.about-summary {
  padding-left: 12px;
  padding-right: 12px;
  font-size: 14px;
}

.about-top {
  width: 52%;
  margin: 0 auto;
}