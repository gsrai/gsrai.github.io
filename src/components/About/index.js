import React from 'react'
import PropTypes from 'prop-types'

import './about.css'

export default class About extends React.Component {
  render() {
    return (
      <div className='row flex-center'>
        <div className='card' style={{ width: '20rem', marginBottom: '2rem' }}>
          <img src='assets/profilepic.jpeg' alt='profile picture' />
          <div className='card-body'>
            <div className='row flex-center'>
              <h5>{this.props.name}, {this.props.age}</h5>
              <h5>{this.props.occupation} in {this.props.location}</h5>
            </div>
            <p className='card-text project-desc'>
              Hi, My name is Gagondeep and I am a Software Developer living in London.
              <br />
              <br />
              I specialize in Java and Javascript FullStack development, but in my spare
              time I enjoy learning new technologies and concepts like the blockchain and machine learning.
              <br />
              <br />
              Recently in my own time I have been using Javascript to create a few Apps such as a
              mobile game in React Native and a chrome extension in React & Redux. Please check my Github account below
              to see my open source projects.
            </p>
            <div className='project-links'>
              <a className='paper-btn ' href='mailto:gagondeep.srai@gmail.com'>Email</a>
              <a className='paper-btn' href='https://github.com/gsrai'>Github</a>
              <a className='paper-btn' href='https://uk.linkedin.com/pub/gagondeep-srai/97/a22/265'>Linkedin</a>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

About.propTypes = {
  name: PropTypes.string.isRequired,
  age: PropTypes.number.isRequired,
  occupation: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired
}
