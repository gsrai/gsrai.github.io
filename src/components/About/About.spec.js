import { shallow } from 'enzyme'
import React from 'react'
import About from './index'

describe('Component Rendering', () => {
  test('component should render correctly', () => {
    const aboutProps = {
      name: 'John Doe',
      age: 25,
      occupation: 'Software Developer',
      location: 'Mountain View, California'
    }

    const wrapper = shallow(<About {...aboutProps} />)
    expect(wrapper).toMatchSnapshot()
  })
})
