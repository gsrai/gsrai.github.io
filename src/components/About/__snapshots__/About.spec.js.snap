// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Rendering component should render correctly 1`] = `
<div
  className="row flex-center"
>
  <div
    className="card"
    style={
      Object {
        "marginBottom": "2rem",
        "width": "20rem",
      }
    }
  >
    <img
      alt="profile picture"
      src="assets/images/profilepic.jpeg"
    />
    <div
      className="card-body"
    >
      <div
        className="row flex-center"
      >
        <h5>
          John Doe
          , 
          25
        </h5>
        <h5>
          Software Developer
           in 
          Mountain View, California
        </h5>
      </div>
      <p
        className="card-text project-desc"
      >
        Hi, My name is Gagondeep and I am a Software Developer living in London.
        <br />
        <br />
        I specialize in Java and Javascript FullStack development, but in my spare time I enjoy learning new technologies and concepts like the blockchain and machine learning.
        <br />
        <br />
        Recently in my own time I have been using Javascript to create a few Apps such as a mobile game in React Native and a chrome extension in React & Redux. Please check my Github account below to see my open source projects.
      </p>
      <div
        className="project-links"
      >
        <a
          className="paper-btn "
          href="mailto:gagondeep.srai@gmail.com"
        >
          Email
        </a>
        <a
          className="paper-btn"
          href="https://github.com/gsrai"
        >
          Github
        </a>
        <a
          className="paper-btn"
          href="https://uk.linkedin.com/pub/gagondeep-srai/97/a22/265"
        >
          Linkedin
        </a>
      </div>
    </div>
  </div>
</div>
`;
